torch/_functorch/aot_autograd.py

```diff
@@ -1050,9 +1050,52 @@ class AOTConfig:
     num_params_buffers: int
     aot_id: int
 
+def aot_dispatch_base_traced_fn(flat_fn, meta, synthetic_base_info, *flat_args_with_views_handled):
+    f_flat_args_with_views_handled = pytree.tree_map(to_fun, flat_args_with_views_handled)
+    torch._enable_functionalization(reapply_views=True)
+    try:
+        f_flat_args = unpack_synthetic_bases(f_flat_args_with_views_handled, synthetic_base_info)
+        f_outs = flat_fn(*f_flat_args)
+    finally:
+        torch._disable_functionalization()
+
+    mutated_inputs_to_return = [
+        x
+        for (i, x) in enumerate(f_flat_args)
+        if meta.input_info[i].mutates_data or meta.input_info[i].mutates_metadata
+    ]
+    all_outs = *mutated_inputs_to_return, *f_outs
+
+    return pytree.tree_map(from_fun, all_outs)
 
 def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig):
-    fw_module = make_fx(flat_fn, aot_config.decompositions)(*flat_args)
+    meta, _out = run_functionalized_fw_and_collect_metadata(flat_fn)(
+        *flat_args
+    )
+
+    _num_outputs = len(meta.output_info)
+    _num_mutated_inputs = len(
+        [x for x in meta.input_info if x.mutates_data or x.mutates_metadata]
+    )
+    _input_info = meta.input_info
+
+    flat_args_with_views_handled, _synthetic_base_info = merge_view_inputs(
```

Comment:
Remind me why we have to merge view inputs for the non-autograd case?

Reply:
yep - if you have this code:
```
def f(x, y):
    x.add_(1)
    return x + y

inp = torch.ones(...)
f(inp, inp.view(-1)
```

Then in order for functionalization to pick up the fact that x and y are aliased (and thus the mutation on x should also affect y), we need the graph that we trace to take a synthetic base as an input, that functionalization then sees us take views off of to create the actual inputs.

I guess the one caveat is that the "synthetic base" logic is a lot simpler in the non-autograd case, because we don't have to worry about handling non-differentiable views of differentiable bases. But that just means that all of the cases for synthetic bases should be handled today, in the non-autograd case.